core/domain/collection_domain.py

```diff
@@ -383,23 +383,8 @@ def serialize(self):
             str. JSON-encoded string encoding all of the information composing
             the object.
         """
-        collection_dict = {
-            'id': self.id,
-            'title': self.title,
-            'category': self.category,
-            'objective': self.objective,
-            'language_code': self.language_code,
-            'tags': self.tags,
-            'schema_version': self.schema_version,
-            'nodes': [
-                node.to_dict() for node in self.nodes
-            ],
-            'version': self.version,
-            'created_on': utils.convert_naive_datetime_to_string(
-                self.created_on),
-            'last_updated': utils.convert_naive_datetime_to_string(
-                self.last_updated)
-        }
+        collection_dict = self.to_dict()
+        collection_dict['version'] = self.version
```

Comment:
It's actually a bit odd that to_dict() doesn't include the version. If you like, you can add it there; ditto elsewhere for other object types. (I don't see any harm in doing so.) That would probably make things cleaner.

Just double-check the call-sites to make sure it's not being set twice. Maybe also add backend tests to ensure that the only differences between the dict version and the serialized version is always only last_updated and created_on, to prevent skew in the future.

Reply:
Ah I see, thanks for explaining. In that case, please add a comment above each "['version'] = self.version" statement to explain why we do it like this ("The only reason we do this is so that ...") but also add a note saying "If this is changed in the future, the following will need to be done to all YAML files: .."). That should provide enough context and we can then punt the decision.

Reply:
Done.